spec/models/reducer_spec.rb

```diff
@@ -66,7 +66,7 @@ def reduce_into(extracts, reductions=nil)
     reducer= build :reducer
 
     extract_fetcher = instance_double(ExtractFetcher, extracts: extracts)
-    reduction_fetcher = instance_double(ReductionFetcher, retrieve: SubjectReduction, has_expired?: false)
+    reduction_fetcher = instance_double(ReductionFetcher, retrieve: SubjectReduction.new, has_expired?: false)
```

Comment:
Was this the only change necessary to test the different behaviour added by the `in_place` reduction_fetcher logic ?

Reply:
In this PR I changed the semantics of the `ReductionFetcher#where` method: where it used to return an `ActiveRelation` which could be queried with `#first_or_initialize`, having to iterate the prefetched reduction objects obliges us to return an object directly (or construct one).